airflow/contrib/executors/kubernetes_executor.py

```diff
@@ -340,15 +356,34 @@ def _run(self, kube_client, resource_version, worker_uuid, kube_config):
         watcher = watch.Watch()
 
         kwargs = {'label_selector': 'airflow-worker={}'.format(worker_uuid)}
+
         if resource_version:
             kwargs['resource_version'] = resource_version
+
         if kube_config.kube_client_request_args:
             for key, value in kube_config.kube_client_request_args.items():
                 kwargs[key] = value
 
-        last_resource_version = None
-        for event in watcher.stream(kube_client.list_namespaced_pod, self.namespace,
-                                    **kwargs):
+        if resource_version:
+            last_resource_version = resource_version
+        else:
+            last_resource_version = None
```

Comment:
What/where is this used?

Reply:
Oh I see, in the unchanged code.

Reply:
I was concerned about handling the exception and allowing last_resource_version to be returned as None, even if the incoming resource_version was > 0